install.sh

```diff
@@ -233,7 +233,18 @@ nvm_detect_profile() {
   local DETECTED_PROFILE
   DETECTED_PROFILE=''
 
-  if [ -n "${BASH_VERSION-}" ]; then
+  # Detect the user's login shell
+  local DETECTED_SHELL
+  local SHELLRC
+
+  DETECTED_SHELL="${SHELL##*/}"
+  SHELLRC="$HOME/.${DETECTED_SHELL}rc"
```

Comment:
how does this affect bashrc vs bash_profile, zshrc vs zprofile, etc?

Reply:
So with this, the current logic reads:
1. Get the user's shell
2. If they have .bashrc or .zshrc, use that
3. Otherwise, check if BASH_VERSION is set
4. If yes, check for .bashrc, then .bash_profile

The question is: do we want mirror logic with DETECTED_SHELL. I'm taking advantage of a single logical branch: trying SHELLRC, which is likely most common, then iterating through various other files. However, [based on this note about the $SHELL variable](http://tldp.org/LDP/abs/html/internalvariables.html#AMIROOT), it's not builtin and I'm not sure if it's a system guarantee.

I'm not really sure how thorough we want to be here, what do you think?

Reply:
I definitely think we can't rely on $SHELL being present. I'm also not sure if `.dashrc` or `.kshrc` are even things.

This logic should probably be as verbose as needed to be very clear about what it's doing and in what order. I'd like to be as thorough as possible, and in particular, I don't want to break anything that's currently working.